crates/optimism/storage/src/chain.rs

```diff
@@ -0,0 +1,124 @@
+use alloy_consensus::Header;
+use alloy_primitives::BlockNumber;
+use core::marker::PhantomData;
+use reth_chainspec::{ChainSpecProvider, EthChainSpec, EthereumHardforks};
+use reth_db_api::{cursor::DbCursorRO, tables, transaction::DbTx};
+use reth_optimism_forks::OpHardforks;
+use reth_optimism_primitives::OpTransactionSigned;
+use reth_primitives_traits::{Block, BlockBody, FullBlockHeader, SignedTransaction};
+use reth_storage_api::{errors::ProviderResult, DBProvider, StorageLocation};
+
+/// Trait that implements how block bodies are written to the storage.
+#[auto_impl::auto_impl(&, Arc)]
+pub(crate) trait BlockBodyWriter<Provider, Body: BlockBody> {
```

Comment:
I don't think we need to duplicate those traits?

can we not import:

https://github.com/paradigmxyz/reth/blob/c40d059dd37b2f4cf780ff00782b4ead44e95fe4/crates/storage/storage-api/src/chain.rs#L65-L65